### ISSUE TITLE
order of function eval bug

### ISSUE THREAD

**Issue by jread-usgs** [USGS-VIZLAB/water-use-15#268]
I'm not sure I have the expertise to figure this out, and I am also not sure what impact (if any) this has on the viz, but we often get an error on page load:

![image](https://user-images.githubusercontent.com/2349007/39361234-20d3a550-49e7-11e8-8686-b632540fc90e.png)
within `updateLegendTextToView()`
here:
```js
var state_data = waterUseViz.stateData
      .filter(function(d) { 
        return d.abrv === activeView; 
    });
```
(the error is `map_utils.js?_c=7941337:301 Uncaught TypeError: waterUseViz.stateData.filter is not a function`)

**Comment by ldecicco-USGS** [comment]
Gah, that means the async loading still isn't in the right order. I've learned that that error means it still has null for the stateData.  I haven't seen that error since my big pr a couple days ago....but that was the bulk of my frustration for the legend task. I'll keep thinking about it, and maybe one of our D3 experts will realize why it's still loading wrong. Do you happen to know if the error only comes when the browser doesn't have a hash? Or is it just once in awhile?

**Comment by aappling-usgs** [comment]
I thought I had fixed this with the `if()` statements at https://github.com/USGS-VIZLAB/water-use-15/blob/master/js/build_map.js#L217 and https://github.com/USGS-VIZLAB/water-use-15/blob/master/js/build_map.js#L229, and I can't reproduce it right now, but I see that those commits happened 5 days ago whereas this issue was created 2 days ago, and I was able to reproduce it on some branch, some build, details now lost, this morning. So I'm not sure whether those changes fully fix the problem.

**Comment by lindsaycarr** [comment]
I'm getting this consistently when the URL I go to is for a state rather than loading the US first and then clicking the state. It seems to be because `addStates` ends up calling updating the view (which eventually calls `updateLegendTextToView` but `waterUseViz.stateData` is loaded later in the `fillMap` function. 

`wu_state_data.json` and `wu_data_15_sum.json` are really small files so I think this can just be fixed by having those part of the initial queue. I'm going to try that.

**Comment by lindsaycarr** [comment]
This was fixed in #286.
